readthedocs/api/v2/views/integrations.py

```diff
@@ -379,6 +387,18 @@ def handle_webhook(self):
                 raise ParseError('Parameter "ref" is required')
         # Sync versions on other PUSH events that create or delete
         elif event in (GITHUB_CREATE, GITHUB_DELETE, GITHUB_PUSH):
+            if event == GITHUB_PUSH:
+                # GitHub will send push and create/delete events on a creation/deletion.
+                # If we receive a push event we need to check if the webhook doesn't
+                # already have the create/delete events. So we don't trigger the sync twice.
+                # We listen to push events for creation/deletion for old webhooks only.
```

Comment:
Good comment 👍

Reply:
I don't follow this code :cry: 

We are already handling GITHUB_PUSH in the previous if with `deleted or created` (line 382), why don't we just remove the GITHUB_PUSH from this `elif` (line 389) instead? That seems to do the same without extra code.

Like this on line 389:

```
elif event in (GITHUB_CREATE, GITHUB_DELETE):
    return self.sync_versions(self.project)
```

That way, we are only syncing versions on CREATE and DELETE, but not on regular PUSH.

Reply:
I think what we want is:

* `GITHUB_PUSH and (created or deleted)` -> sync versions
* `GITHUB_CREATE or GITHUB_DELETE` -> sync versions
* all the other possibilities -> do not sync versions

If this is correct, I do see this logic easier to understand than our current implementation.

Reply:
> We are already handling GITHUB_PUSH in the previous if with deleted or created 

We are triggering a build in that case, and we check for `not (deleted or created)`

> why don't we just remove the GITHUB_PUSH from this elif (line 389) instead?

We want to listen to PUSH events for sync, but only trigger a sync if the webhook isn't subscribed to creation/deletion events.

Reply:
GitHub will send two events when a branch/tag is created/deleted (if the webhook is listening to creation/deletion events), that's why we need to discard the push event in that case.

> GITHUB_PUSH and (created or deleted) -> sync versions

That will still need to check if the webhook isn't subscribed to creation/deletion. Otherwise it'll trigger two syncs.

> GITHUB_CREATE or GITHUB_DELETE -> sync versions

This will trigger a second sync for users that have webhooks listening to creation/deletion events.